glue/flagging_script_glue/flagging.py

```diff
@@ -767,131 +742,97 @@ def z_normalize_groupby(s: pd.Series):
-            & (df["sv_pricing"].str.contains("High")),
-            (df["sv_anomaly"] == "Outlier") & (df["sv_pricing"].str.contains("High")),
-            (df["sv_pricing"].str.contains("High price swing")),
+        # Define conditions for price-based reasons
+        price_conditions = [
             (df["sv_pricing"].str.contains("High"))
             & (df["sv_which_price"] == "(raw & sqft)"),
-            (df["sv_pricing"].str.contains("High")) & (df["sv_which_price"] == "(raw)"),
-            (df["sv_pricing"].str.contains("High"))
-            & (df["sv_which_price"] == "(sqft)"),
-            (df["sv_short_owner"] == "Short-term owner")
-            & (df["sv_pricing"].str.contains("Low")),
-            (df["sv_name_match"] != "No match")
-            & (df["sv_pricing"].str.contains("Low")),
-            (
-                df[["sv_buyer_category", "sv_seller_category"]]
-                .eq("legal_entity")
-                .any(axis=1)
-            )
-            & (df["sv_pricing"].str.contains("Low")),
-            (df["sv_anomaly"] == "Outlier") & (df["sv_pricing"].str.contains("Low")),
-            (df["sv_pricing"].str.contains("Low price swing")),
             (df["sv_pricing"].str.contains("Low"))
             & (df["sv_which_price"] == "(raw & sqft)"),
-            (df["sv_pricing"].str.contains("Low")) & (df["sv_which_price"] == "(raw)"),
+            (df["sv_pricing"].str.contains("High") & (df["sv_which_price"] == "(raw)")),
+            (df["sv_pricing"].str.contains("Low") & (df["sv_which_price"] == "(raw)")),
+            (df["sv_pricing"].str.contains("High"))
+            & (df["sv_which_price"] == "(sqft)"),
             (df["sv_pricing"].str.contains("Low")) & (df["sv_which_price"] == "(sqft)"),
         ]
 
-        labels = [
-            "Home flip sale (high)",
-            "Family sale (high)",
-            "Non-person sale (high)",
-            "Anomaly (high)",
-            "High price swing",
-            "High price (raw & sqft)",
-            "High price (raw)",
-            "High price (sqft)",
-            "Home flip sale (low)",
-            "Family sale (low)",
-            "Non-person sale (low)",
-            "Anomaly (low)",
-            "Low price swing",
-            "Low price (raw & sqft)",
-            "Low price (raw)",
-            "Low price (sqft)",
+        # Define labels for price-based reasons
+        price_labels = [
+            "sv_ind_price_high_price_raw_and_sqft",
+            "sv_ind_price_low_price_raw_and_sqft",
+            "sv_ind_price_high_price",
+            "sv_ind_price_low_price",
+            "sv_ind_price_high_price_sqft",
+            "sv_ind_price_low_price_sqft",
         ]
 
-    df["sv_outlier_type"] = np.select(conditions, labels, default="Not outlier")
-
-    return df
+    combined_conditions = price_conditions + char_conditions
+    combined_labels = price_labels + char_labels
 
+    outlier_type_dict = dict(zip(combined_labels, combined_conditions))
```

Comment:
**[Nitpick, non-blocking]** With my previous comment (https://github.com/ccao-data/model-sales-val/pull/128#discussion_r1627990344), I was imagining that instead of zipping these lists and casting them to a dict, we might just combine them into a single data structure at the point of definition; that way we wouldn't need to bother with this extra step, and it would be easier for readers to quickly understand which conditions map to which labels without having to compare two lists by index. That being said, I don't think it's actually a big deal, and I'm fine going forward with whatever approach you prefer -- I just want to make sure the intention of my comment was clear! If we decide to move forward with two lists that we zip together, as in this implementation, I don't think the extra `dict()` call is necessary since the zipped lists can be iterated on their own.

Reply:
I removed the dict creation and just zipped them for now. This function is at least a lot more organized and concise than the previous iteration.